### PR TITLE
control-service: run release  test on dependency version change

### PR DESCRIPTION
**Pull request by tozka**
In case of change in versions-of-external-dependencies.gradle we are directly making a new release without triggering release acceptance test which may result in broken release. It's better to make sure the release is tested. 

Also removed job-builder changes from trigger deployment and release test for now as generally default job-builder version is pinned so when changed it would trigger new test. We need to devise a way to test all differnt job-builders anyhow. 

Finally refactored slighlty the yaml anchors to re-use previously defined ones.

Testing Done: this PR

### CODE REVIEW

**Review by murphp15 — APPROVED** [review]

**Review by doks5 — APPROVED** [review]
Looks like the yaml is broken.

File: projects/control-service/cicd/.gitlab-ci.yml [hunk 1]
```diff
@@ -339,4 +322,5 @@ control_service_vdk_heartbeat_release:
   only:
     refs:
       - main
-    changes: *control_service_change_locations
```
Comment (murphp15) [inline projects/control-service/cicd/.gitlab-ci.yml:342]: Why did you not just change control_service_change_locations to be the union of control_service_code_change_locations and control_service_helm_change_locations?

Comment (tozka) [reply projects/control-service/cicd/.gitlab-ci.yml:342]: Do you know how ?

From what I've found Merging arrays is not really supported in YAML https://github.com/yaml/yaml/issues/35 
Nor in gitlab references .

Comment (murphp15) [reply projects/control-service/cicd/.gitlab-ci.yml:342]: Can changes take a list then instead of declaring the property twice?

Comment (tozka) [reply projects/control-service/cicd/.gitlab-ci.yml:342]: I changed it to use rules instead

Comment (murphp15) [reply projects/control-service/cicd/.gitlab-ci.yml:342]: now there is lots of duplication. 
if there is no way of getting rid of the duplication then I'm still happy for it to be merged.

Comment (tozka) [reply projects/control-service/cicd/.gitlab-ci.yml:342]: I will do it in subsequent PR. I want to make sure the CI is working functionally correct. Then I'd refactor it.

File: projects/control-service/cicd/.gitlab-ci.yml [hunk 2]
```diff
@@ -267,13 +246,14 @@ control_service_deploy_testing_data_pipelines:
...
     - bash -ex ./projects/control-service/cicd/deploy-testing-pipelines-service.sh
   retry: !reference [.control_service_retry, retry_options]
-  only:
-    refs:
-      - main
-    changes: *control_service_deploy_testing_changes
-  except:
-    changes:
-      - projects/control-service/projects/helm_charts/pipelines-control-service/version.txt
+  rules:
```
[the remark this answers is not in the document]
Comment (tozka) [reply projects/control-service/cicd/.gitlab-ci.yml:249]: I will do it in subsequent PR. I want to make sure the CI is working functionally correct. Then I'd refactor it.

File: projects/control-service/cicd/.gitlab-ci.yml (hunk 1 above)
Comment (murphp15) [reply projects/control-service/cicd/.gitlab-ci.yml:342]: Cool

File: projects/control-service/cicd/.gitlab-ci.yml [hunk 3]
```diff
@@ -5,20 +5,7 @@
       - always
 
 
-# This grouping represents ALL files related to the control plane.
-# This grouping triggers actions like new releases etc. but don't trigger unit or integration tests.before_script
-# because it also includes the helm chart folders and we don't want to run unit and integration tests anytime
-# the helm chart changes
-.control_service_changes: &control_service_change_locations
```
Comment (1singh1) [inline projects/control-service/cicd/.gitlab-ci.yml:12]: Change it to

**Pull request merged**